query.go

```diff
@@ -346,6 +346,7 @@ type RecordAccess struct {
 	record interface{}
 	field  interface{}
 	query  *Query
+	store  *Store
```

Comment:
this was the only tricky bit; since RecordAccess is passed in callbacks to user code, we have to keep the encode/decode funcs somewhere here.